sources/imdsclient/src/lib.rs

```diff
@@ -21,20 +21,27 @@ use log::{debug, info, trace, warn};
 use reqwest::Client;
 use serde_json::Value;
 use snafu::{ensure, OptionExt, ResultExt};
-use std::time::Duration;
-use tokio::time;
+use tokio::time::{sleep, timeout, Duration};
+use tokio_retry::{strategy::FibonacciBackoff, Retry};
 
 const BASE_URI: &str = "http://169.254.169.254";
 const PINNED_SCHEMA: &str = "2021-01-03";
 
 // Currently only able to get fetch session tokens from `latest`
 const SESSION_TARGET: &str = "latest/api/token";
 
+fn retry_token_strategy() -> impl Iterator<Item = Duration> {
+    FibonacciBackoff::from_millis(1000)
+        .factor(3)
```

Comment:
Why 3?  I don't think the docs or code around factor are very clear, but this seems to me to sort of arbitrarily multiply the result by 3..?  So if you have `FibonacciBackoff::from_millis(1000)`, you'd actually start by waiting 3s, then 3s, 6s, 9s, etc.  Why not 1s, 1s, 2s, 3s, etc.?  I guess this is similar to my suggestion to move down to 500ms; I don't see a reason to wait very long (in the beginning) for a local service that's in the boot path.

Reply:
I think I was being overly cautious as to not spam IMDS. I dropped to 500ms and went with a factor of 1 with a max delay of 10 to keep things simple.